### PR TITLE
Remove dependency on node log output for detecting startup success

### PR DESCRIPTION
**Pull request by hiddentao**
As the title says. We now wait 4 seconds after starting the node - if no error is returned within that time then we assume the node has started up successfully and continue onto attempting the IPC connection.

### CODE REVIEW

File: modules/ethereumNode.js [hunk 1]
```diff
@@ -526,19 +544,12 @@ class EthereumNode extends EventEmitter {
...
 
-                        if ('eth' === nodeType) {
-                            // (eth) prevent started until str appears
-                            if (-1 === dataStr.indexOf('jsonrpc')) {
-                                log.trace('Running eth so wait until we see JSONRPC message');
-
-                                return;
-                            }
-                        } else if ('geth' === nodeType) {
+                        if ('geth' === nodeType) {
```
Comment (frozeman) [inline modules/ethereumNode.js]: I find this way around confusing :)
better is: `nodeType === 'geth'`

or do you have a reason for that?

File: modules/ethereumNode.js [hunk 2]
```diff
@@ -488,6 +488,22 @@ class EthereumNode extends EventEmitter {
 
                 const proc = spawn(binPath, args);
 
+                /*
+                    Assume startup succeeded after 5 seconds. At this point 
+                    IPC connections are usually possible.
+                */
+                setTimeout(() => {
```
Comment (frozeman) [inline modules/ethereumNode.js]: This should only be started, in `.stderr.on('data', ...)`, at least then we know for sure that it at least started

File: modules/ethereumNode.js (hunk 1 above)
Comment (hiddentao) [inline modules/ethereumNode.js]: Defensive programming. If by mistake `=` is used it won't compile. Whereas if you did `nodeType = "geth"` that still compiles and will produce an undesired outcome.

File: modules/ethereumNode.js (hunk 2 above)
Comment (hiddentao) [inline modules/ethereumNode.js]: Agreed. I've changed it so that as soon as any data is received (stdout or stderr) it starts the timer.

**Comment by alexvandesande** [comment]
Worked for me without a glitch

**Pull request merged**